esx_service/vmdk_ops.py

```diff
@@ -708,7 +695,10 @@ def executeRequest(vm_uuid, vm_name, config_path, cmd, full_vol_name, opts):
     Returns None (if all OK) or error string
     """
     logging.debug("config_path=%s", config_path)
-    vm_datastore = get_datastore_name(config_path)
+    error_info, vm_datastore = vmdk_utils.get_datastore_name_from_config_path(config_path)
```

Comment:
There is no "datastore" name in the "config_path". Only "datastore_url" in the config_path. We tried to figure out the datastore name from datastore_url. If we don't refresh the cache, for the datastore_url, it still map to old datastore name.